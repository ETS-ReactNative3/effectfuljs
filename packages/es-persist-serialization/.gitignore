index.js
node_modules
tmp-*
*.orig
*.log
*~
*.map
.DS_Store
.tern-port
err-log.json
