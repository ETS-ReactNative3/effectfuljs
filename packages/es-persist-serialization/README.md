# @effectful/es-persist-serialization

The library uses [@effectful/es-persist](https://github.com/awto/effectfuljs/tree/master/packages/serialization)
to store async function state for
[@effectful/es-persist](https://github.com/awto/effectfuljs/tree/master/packages/es-persist).

It re-exports both of the libraries, and `@effectful/serialization` is
imported as peer dependency.

## Usage

Install from npm:

```
$ npm install --save-dev @effectful/es
$ npm install --save @effectful/es-persist-serialization @effectful/serialization
```

In .babelrc:

```json
{
  "presets": ["@effectful/es-persist-serialization/transform"]
}

```

If there are other babel presets, add `"passPerPreset":true`.

## License

Distributed under the terms of the [The MIT License (MIT)](LICENSE). 

