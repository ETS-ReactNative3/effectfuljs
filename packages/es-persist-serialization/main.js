import * as S from "@effectful/serialization";
import * as R from "@effectful/es-persist";
export {
  generator,
  generatorFunction,
  iterator,
  iteratorM,
  forInIterator,
  contextSymbol,
  sendSymbol,
  awaitSymbol,
  context,
  async,
  asyncGenerator,
  lock,
  all,
  any,
  producer,
  current,
  idle,
  abort,
  managed
} from "@effectful/es-persist";

export {
  read,
  write,
  regConstructor,
  regOpaqueObject,
  bind,
  descriptorSymbol,
  parse,
  stringify,
  regDescriptor,
  regOpaquePrim
} from "@effectful/serialization";

R.constructors.forEach(v => S.regConstructor(v));
export const asyncFunction = R.makeAsyncFunctionConstructor(S.regConstructor);
export const asyncGeneratorFunction = R.makeAsyncGeneratorFunctionConstructor(
  S.regConstructor
);
