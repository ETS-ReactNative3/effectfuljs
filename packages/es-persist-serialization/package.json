{
  "name": "@effectful/es-persist-serialization",
  "version": "1.3.6",
  "description": "@effectful/es-persist with @effectful/serialization for storing state",
  "main": "index.js",
  "module": "main.js",
  "jsnext:main": "main.js",
  "scripts": {
    "test": "mocha",
    "build": "babel --config-file ../es/babel.cjs.json -o index.js main.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/awto/effectfuljs.git"
  },
  "keywords": [
    "effectful",
    "effectfuljs",
    "monad",
    "linq"
  ],
  "author": "Vitaliy Akimov <vitaliy.akimov@gmail.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/awto/effectfuljs/issues"
  },
  "homepage": "https://github.com/awto/effectfuljs#readme",
  "peerDependencies": {
    "@effectful/serialization": "^1.0.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.0.0",
    "@effectful/serialization": "^1.3.6",
    "mocha": "^5.0.5"
  },
  "dependencies": {
    "@effectful/es-persist": "^1.3.1"
  },
  "gitHead": "d3b25d1148c0b8da0bb45a6a52b8b8cac3150ca5"
}
