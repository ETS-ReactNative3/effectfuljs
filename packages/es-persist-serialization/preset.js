module.exports = require("@effectful/es-persist/preset")
  .options({importRT:"@effectful/es-persist-serialization"})
