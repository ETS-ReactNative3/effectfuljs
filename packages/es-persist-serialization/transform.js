module.exports = require("@effectful/es-persist/transform").options({
  importRT: "@effectful/es-persist-serialization"
});
