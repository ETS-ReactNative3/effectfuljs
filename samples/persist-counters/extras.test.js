import * as R from "@effectful/es-persist-serialization";

import { incrementAsync, incrementIfOdd } from "./extras";

function extrasDispatch() {}
R.regOpaqueObject(extrasDispatch);

test("incrementIfOdd", async () => {
  expect.assertions(2);
  for await (const i of incrementIfOdd(
    [{ type: "VALUE", value: 1 }, { type: "VALUE", value: 2 }],
    extrasDispatch
  )) {
    if (i.type === "MENU_ITEM") expect(R.write(i)).toMatchSnapshot();
  }
});

test("incrementAsync", async () => {
  expect.assertions(1);
  for await (const i of incrementAsync([], extrasDispatch)) {
    if (i.type === "MENU_ITEM") expect(R.write(i)).toMatchSnapshot();
  }
});
