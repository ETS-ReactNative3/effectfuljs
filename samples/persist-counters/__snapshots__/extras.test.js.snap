// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incrementAsync 1`] = `
Object {
  "f": Array [
    Array [
      "type",
      "MENU_ITEM",
    ],
    Array [
      "index",
      300,
    ],
    Array [
      "value",
      Object {
        "$": "ReactElement",
        "children": Array [
          "Increment Async",
        ],
        "props": Object {
          "f": Array [
            Array [
              "onClick",
              Object {
                "r": 0,
              },
            ],
            Array [
              "children",
              Object {
                "$": "undefined",
              },
            ],
          ],
        },
        "type": "button",
      },
    ],
  ],
  "x": Array [
    Object {
      "$": "Bind",
      "args": Array [
        Object {
          "$": "extrasDispatch",
        },
      ],
      "f": Array [
        Array [
          "prototype",
          Object {
            "f": Array [
              Array [
                "constructor",
                Object {
                  "r": 0,
                },
                2,
              ],
            ],
          },
          3,
        ],
      ],
      "func": Object {
        "$": "runIncrementAsync",
      },
      "self": Object {
        "$": "undefined",
      },
    },
  ],
}
`;

exports[`incrementIfOdd 1`] = `
Object {
  "f": Array [
    Array [
      "type",
      "MENU_ITEM",
    ],
    Array [
      "index",
      400,
    ],
    Array [
      "value",
      Object {
        "$": "ReactElement",
        "children": Array [
          "Increment if odd",
        ],
        "props": Object {
          "f": Array [
            Array [
              "onClick",
              Object {
                "r": 0,
              },
            ],
            Array [
              "children",
              Object {
                "$": "undefined",
              },
            ],
          ],
        },
        "type": "button",
      },
    ],
  ],
  "x": Array [
    Object {
      "$": "Bind",
      "args": Array [
        Object {
          "$": "extrasDispatch",
        },
        "INCREMENT",
        Object {
          "$": "undefined",
        },
      ],
      "f": Array [
        Array [
          "prototype",
          Object {
            "f": Array [
              Array [
                "constructor",
                Object {
                  "r": 0,
                },
                2,
              ],
            ],
          },
          3,
        ],
      ],
      "func": Object {
        "$": "_signal",
      },
      "self": null,
    },
  ],
}
`;

exports[`incrementIfOdd 2`] = `
Object {
  "f": Array [
    Array [
      "type",
      "MENU_ITEM",
    ],
    Array [
      "index",
      400,
    ],
    Array [
      "value",
      Object {
        "$": "ReactElement",
        "children": Array [
          "Increment if odd",
        ],
        "props": Object {
          "f": Array [
            Array [
              "onClick",
              Object {
                "$": "nop",
              },
            ],
            Array [
              "children",
              Object {
                "$": "undefined",
              },
            ],
          ],
        },
        "type": "button",
      },
    ],
  ],
}
`;
